include/AudioBufferView.h

```diff
@@ -0,0 +1,256 @@
+/*
+ * AudioBufferView.h - Non-owning views for interleaved and
+ *                     non-interleaved (planar) buffers
+ *
+ * Copyright (c) 2025 Dalton Messmer <messmer.dalton/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_AUDIO_BUFFER_VIEW_H
+#define LMMS_AUDIO_BUFFER_VIEW_H
+
+#include <cassert>
+#include <span>
+#include <type_traits>
+
+#include "LmmsTypes.h"
+
+namespace lmms
+{
+
+//! Use when the number of channels is not known at compile time
+inline constexpr auto DynamicChannelCount = static_cast<proc_ch_t>(-1);
+
+
+namespace detail {
+
+// For static channel count
+template<typename SampleT, proc_ch_t channelCount>
+class BufferViewData
+{
+public:
+	constexpr BufferViewData() = default;
+	constexpr BufferViewData(const BufferViewData&) = default;
+
+	constexpr BufferViewData(SampleT* data, proc_ch_t channels, f_cnt_t frames) noexcept
+		: m_data{data}
+		, m_frames{frames}
+	{
+		assert(channels == channelCount);
+	}
```

Comment:
Why have this constructor if the channel count was already determined at compile time?